vital/tests/test_metadata.cxx

```diff
@@ -0,0 +1,123 @@
+/*ckwg +29
+ * Copyright 2020 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <vital/types/metadata.h>
+#include <vital/types/metadata_traits.h>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <stdint.h>
+
+using namespace ::kwiver::vital;
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+TEST ( metadata, typed_metadata )
+{
+  // create item
+  std::unique_ptr< typed_metadata > tmds = new typed_metadata( VITAL_META_METADATA_ORIGIN, "item data" );
+  auto* tmdd = new typed_metadata< VITAL_META_PLATFORM_HEADING_ANGLE, double >(
+    std::string( "test double item" ), 3.14159 );
+  auto* tmdi = new typed_metadata< VITAL_META_UNIX_TIMESTAMP, uint64_t >(
+    std::string( "test uint item" ), uint64_t( 314159 ) );
```

Comment:
```suggestion
  auto tmds = typed_metadata< VITAL_META_METADATA_ORIGIN, std::string >{
    "test string item", "item data" );
  auto tmdd = typed_metadata< VITAL_META_PLATFORM_HEADING_ANGLE, double >{
    "test double item", 3.14159 };
  auto tmdi = typed_metadata< VITAL_META_UNIX_TIMESTAMP, uint64_t >{
    "test uint item", uint64_t{ 314159 } };
```

Reply:
That said, I'm wondering if this test would benefit from a helper function to create the appropriate `unique_ptr` a la what `metadata_map::add` does. (IOW, what `NEW_METADATA_ITEM` did... I removed it as this test is the only place where that is directly useful, but having a replacement function in this test would make sense.)

Note that such a helper would be particularly helpful in the next test case.